### PR TITLE
Various CI fixes and cleanup

### PR DESCRIPTION
**Pull request by mehrdadn**
## Why are these changes needed?

Addresses various CI issues.

## Related issue number

#631

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: thirdparty/patches/redis-quiet.patch [hunk 1]
```diff
@@ -0,0 +1,49 @@
+diff --git deps/jemalloc/configure deps/jemalloc/configure
```
[the remark this answers is not in the document]
Comment (simon-mo) [reply thirdparty/patches/redis-quiet.patch:1]: the question is more of "how likely will that happen"

[the remark this answers is not in the document]
Comment (mehrdadn) [reply thirdparty/patches/redis-quiet.patch:1]: Good question!  think fixing it in the event of a breakage shouldn't need more than 5 minutes—it's just a matter of copy/pasting a few changed lines.

Note that we don't just update Redis randomly. We have _much_ more difficult patches to take care of for Windows compatibility and such (see the existing patches). So I think the overhead is negligible. And it's also a really significant gain in terms of how much noise it cuts down on in the builds (that itself wastes a lot of time). And in any case this patch is trivial enough to comment out if it somehow becomes a problem.

File: ci/travis/install-bazel.sh [hunk 2]
```diff
@@ -42,21 +42,20 @@ esac
...
-  rm -f "${target}"
+sudo=sudo
+exe_suffix=""
+bazel_target="/usr/local/bin/bazel"
+if [ "${OSTYPE}" = msys ]; then
+  sudo=""
+  exe_suffix=".exe"
+  bazel_target="${MINGW_PREFIX}/bin/bazel.exe"
+fi
+${sudo} curl -f -s -L -R -o "${bazel_target}" "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel-${version}-${platform}-${achitecture}${exe_suffix}"
```
[the remark this answers is not in the document]
Comment (mehrdadn) [reply ci/travis/install-bazel.sh:53]: Ohh great catch!! I'll fix it, thanks.

File: ci/travis/install-dependencies.sh [hunk 3]
```diff
@@ -60,7 +60,7 @@ install_miniconda() {
       ;;
   esac
   reload_env
-  if [ -n "${PYTHON-}" ]; then
+  if [ -n "${PYTHON-}" ] && [ "${PYTHON}" != "$(python -s -c "import sys; sys.stdout.write('.'.join(map(str, sys.version_info[:2])))")" ]; then
```
[the remark this answers is not in the document]
Comment (mehrdadn) [reply ci/travis/install-dependencies.sh]: Oh this line already changed in a subsequent commit—I think it's a bit clearer there, let me know if it still needs changing?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -2,6 +2,11 @@ name: CI
 
 on: [push, pull_request]
 
+env:
+  GITHUB_BASE_SHA: ${{ github.event.pull_request.base.sha }}
+  GITHUB_HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+  GITHUB_PULL_REQUEST: ${{ github.event.number }}
+
```
Comment (simon-mo) [inline .github/workflows/main.yml:9]: Can you add a comment saying where these are used?

File: .travis.yml [hunk 5]
```diff
@@ -93,12 +96,13 @@ matrix:
...
-      before_install:
-        - . ./ci/travis/ci.sh init
       install:
+        - . ./ci/travis/ci.sh init
+      before_script:
+        - . ./ci/travis/ci.sh lint
         - . ./ci/travis/ci.sh build
       script:
-        - . ./ci/travis/ci.sh lint
+        - true
```
Comment (simon-mo) [inline .travis.yml:105]: if we are just skipping, we can just remove the script block right?

File: ci/travis/ci.sh [hunk 6]
```diff
@@ -52,21 +50,12 @@ reload_env() {
...
-  fi
-
-  # NOTE: Modifying PATH invalidates Bazel's cache! Do not add to PATH unnecessarily.
-  PATH="${HOME}/miniconda/bin":"${PATH}"
-  if [ "${OSTYPE}" = msys ]; then
-    PATH="${HOME}/miniconda/bin/Scripts":"${PATH}"
+  if [ -z "${TRAVIS_BRANCH-}" ] && [ -n "${GITHUB_WORKFLOW-}" ]; then
+    TRAVIS_BRANCH="${GITHUB_BASE_REF:-${GITHUB_REF}}"
+    TRAVIS_BRANCH="${TRAVIS_BRANCH#refs/heads/}"
+    TRAVIS_BRANCH="${GITHUB_HEAD_SHA:-${TRAVIS_BRANCH}}"  # Need a hash on GitHub Actions when possible (even though on Travis it's always a name)
```
Comment (simon-mo) [inline ci/travis/ci.sh:56]: this comment doesn't say much, can we explain more? 
also can you explain does this block do in the comment?

File: python/ray/tests/test_metrics.py [hunk 7]
```diff
@@ -112,8 +112,9 @@ def f(self):
         ]
         for process in processes:
             # TODO(ekl) why does travis/mi end up in the process list
-            assert ("python" in process or "ray" in process
-                    or "travis" in process)
+            assert ("python" in process or "conda" in process
+                    or "travis" in process or "runner" in process
```
Comment (simon-mo) [inline python/ray/tests/test_metrics.py:116]: runner? why?

**Review by simon-mo — APPROVED** [review]
Few minor nits, great work

File: .github/workflows/main.yml (hunk 4 above)
Comment (mehrdadn) [reply .github/workflows/main.yml:9]: Yup, they're used for translating Travis environment variables, I'll update them.

File: .travis.yml (hunk 5 above)
Comment (mehrdadn) [reply .travis.yml:105]: Sadly not :( if you do that then it falls back to the default outermost block. I think I can technically put `script: []` but it's not clear to me why that works, it's kind of confusing.

File: ci/travis/ci.sh (hunk 6 above)
Comment (mehrdadn) [reply ci/travis/ci.sh:56]: Sure

File: python/ray/tests/test_metrics.py (hunk 7 above)
Comment (mehrdadn) [reply python/ray/tests/test_metrics.py:116]: I think `travis` was meant to refer to the username on Travis, so I added `runner`, which is the username on GitHub Actions.

File: .travis.yml (hunk 5 above)
Comment (simon-mo) [reply .travis.yml:105]: I see. Add a comment so people maintaining this in the future knows?

Comment (mehrdadn) [reply .travis.yml:105]: Sure

**Pull request merged**